### ISSUE TITLE
email_html action raises an error

### ISSUE THREAD

**Issue by fbaligand** [sirensolutions/sentinl#93]
Using the same watcher, I define 2 actions : one `email` and one `email_html` with nearly the same configuration.
I receive well the `email` one, but not the `email_html` one.

When looking at logs, I can see this error log :
`{"type":"log","@timestamp":"2017-01-17T14:50:00Z","tags":["error","Sentinl"],"pid":11701,"message":"An error occurred while executing the watch: TypeError: Cannot read property 'active' of undefined"}`

**Is there a special entry to activate in `kibana.yml` ?**

I use Kibana 4.6.3.

Here's my `kibana.yml` sentinl configuration :
``` yaml
sentinl:
  es:
    timefield: '@timestamp'
    default_index: watcher
    type: watch
    alarm_index: watcher_alarms
  sentinl:
    history: 20
    results: 50
  settings:
    email:
      active: true
      host: localhost
      ssl: false
    report:
      active: true
      tmp_path: /tmp/
```

Here's my watcher : 

``` json
{
  "_index": "watcher",
  "_type": "watch",
  "_id": "test_watch_fba",
  "_score": 1,
  "_source": {
    "disable": false,
    "trigger": {
      "schedule": {
        "later": "every 1 day at 2:27 pm"
      }
    },
    "input": {
      "search": {
        "request": {
          "index": "<logstash-{now-1d/d}>,<logstash-{now-2d/d}>",
          "body": {
            "size": 0,
            "query": {
              "filtered": {
                "query": {
                  "query_string": {
                    "query": "*"
                  }
                },
                "filter": {
                  "range": {
                    "@timestamp": {
                      "from": "now-1d/d",
                      "to": "now-1d/d",
                      "time_zone": "Europe/Paris"
                    }
                  }
                }
              }
            }
          }
        }
      }
    },
    "condition": {
      "script": {
        "script": "payload.hits.total > 0"
      }
    },
    "transform": {},
    "actions": {
      "send_email_fba": {
        "email": {
          "to": "fba@mycompany.com",
          "from": "kibana@mycompany.com",
          "subject": "SENTINL TEST NOT HTML",
          "priority": "high",
          "body": "Total Hits: {{ payload.hits.total }}",
          "stateless": false
        }
      },
      "send_email_html_fba": {
        "email_html": {
          "to": "fba@mycompany.com",
          "from": "kibana@mycompany.com",
          "subject": "SENTINL TEST HTML",
          "priority": "high",
          "body": "Total Hits: {{ payload.hits.total }}",
          "html": "<p>Total Hits: {{ payload.hits.total }}</p>",
          "stateless": false
        }
      }
    }
  }
}
```

**Comment by fbaligand** [comment]
@lmangani 
Is there a special entry to activate in kibana.yml, so that email_html works ?

**Comment by lmangani** [comment]
Anything in the kibana logs when the watcher executes?

**Comment by fbaligand** [comment]
Yes, it is in the issue description.

**Comment by fbaligand** [comment]
@lmangani 
Yes, it is in the issue description :
```
{"type":"log","@timestamp":"2017-01-17T14:50:00Z","tags":["error","Sentinl"],"pid":11701,"message":"An error occurred while executing the watch: TypeError: Cannot read property 'active' of undefined"}
``` 

This error seems to indicate a missing entry in kibana.yml, no ?

**Comment by lmangani** [comment]
Indeed, probably a patch i overlooked - with check it next

**Comment by lmangani** [comment]
spot on: https://github.com/sirensolutions/sentinl/blob/0ebba4e22fe0304660861590d3cc4f0e85caf729/server/lib/actions.js#L203

will patch it shortly

**Comment by fbaligand** [comment]
@lmangani 

I tried to add `email_html.active: true`, but I get this error at kibana startup :
`{"type":"log","@timestamp":"2017-01-18T09:41:58Z","tags":["fatal"],"pid":52875,"level":"fatal","message":"child \"sentinl\" fails because [child \"settings\" fails because [\"email_html\" is not allowed]]","error":{"message":"child \"sentinl\" fails because [child \"settings\" fails because [\"email_html\" is not allowed]]","name":"ValidationError","stack":"ValidationError: child \"sentinl\" fails because [child \"settings\" fails because [\"email_html\" is not allowed]]`

Up to me, one nice fix would be to fix actions.js to check `config.settings.email.active`.
I mean : there is no need to distinct email activation and email_html activation.
Thus, currently, this is not documented, neither supported in configuration.

**Comment by lmangani** [comment]
This was a patch-injected bug most definitely, will resolve it this morning

**Comment by fbaligand** [comment]
Nice news !
Thanks !

**Comment by fbaligand** [comment]
@lmangani 
As the fix is easy to do, I tried the fix on my sentinl installation, and it works great now !

**Comment by fbaligand** [comment]
@lmangani 
By the way, one other thing about email_html : 
- the `html` action field is not used as email body. Only `body` action field is used. HTML tags in `body` field are correctly interpreted.

=> So I think this would be nice to update wiki [here](https://github.com/sirensolutions/sentinl/wiki/SENTINL-Watcher-Actions#email-html) to remove html field and keep only body field with html content. Thus, it would be nice to update also SENTINL SPY so that when we add a `email_html` action, the generated action has no `html` field but a `body` field with html content.

**Comment by lmangani** [comment]
@fbaligand an html email can actually have two bodies, a plain one and an html one attached
```
 emailServer.send({
          text: body,
          from: action.email_html.from,
          to: action.email_html.to,
          subject: subject,
          attachment: [{data: html, alternative: true}]
        }
```

**Comment by lmangani** [comment]
FYI a resync with our private repository will bring this fix to public + build later today

**Comment by lmangani** [comment]
fixed, please confirm if the check if now valid and again thanks for reporting!

**Comment by fbaligand** [comment]
OK, so :
- I tested the fix, and now, it works great ! Thanks a lot for that !
- I noticed that branch 4.6.3 has not been fixed.
- And so, I noticed that there is a branch 4.6.3. For information, why is there is a specific branch for this version ?
- Concerning `html` and `body`, when I define both fields, in may mailbox (which clearly supports HTML mails), I receive only the `body` content (and no attachment).

**Comment by fbaligand** [comment]
@lmangani 
OK, so :
- I tested the fix, and now, it works great ! Thanks a lot for that !
- I noticed that branch 4.6.3 has not been fixed.
- And so, I noticed that there is a branch 4.6.3. For information, why is there is a specific branch for this version ?
- I also noticed that branch 5.0 has not been fixed. Should I open a new issue for that ?
- Concerning `html` and `body`, when I define both fields, in may mailbox (which clearly supports HTML mails), I receive only the `body` content (and no attachment).
